src/psycopmlutils/timeseriesflattener/data_integrity.py

```diff
@@ -0,0 +1,261 @@
+"""Code to generate data integrity and train/val/test drift reports."""
+from pathlib import Path
+from typing import Optional, Tuple
+
+import pandas as pd
+from deepchecks.tabular import Dataset, Suite
+from deepchecks.tabular.checks import (
+    FeatureLabelCorrelation,
+    FeatureLabelCorrelationChange,
+    IdentifierLabelCorrelation,
+    TrainTestLabelDrift,
+)
+from deepchecks.tabular.suites import data_integrity, train_test_validation
+from wasabi import Printer
+
+
+def check_feature_sets_dir(path: Path, nrows: Optional[int] = None) -> None:
+    """Runs Deepcheck data integrity and train/val/test drift checks for a
+    given directory containing train/val/test files. Expects the directory to
+    contain 3 csv files, where the files have ["train", "val", "test"]
+    somewhere in their name.
+
+    The resulting reports are saved to a sub directory as .html files.
+
+    Args:
+        path (Path): Path to a directory containing train/val/test files
+        nrows (Optional[int]): Whether to only load a subset of the data.
+        Should only be used for debugging.
+    """
```

Comment:
Maybe we should refactor so it can handle e.g. only train in the dir? 

Seems likely Andreas' and his students will use this approach :-)